src/WebJobs.Script/Extensions/BindingMetadataExtensions.cs

```diff
@@ -1,12 +1,90 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using Microsoft.Azure.WebJobs.Script.Description;
+using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Azure.WebJobs.Script.Extensions
 {
     public static class BindingMetadataExtensions
     {
+        private const string HttpTriggerKey = "httpTrigger";
+        private const string EventGridTriggerKey = "eventGridTrigger";
+        private const string BlobTriggerKey = "blobTrigger";
+
+        private static readonly HashSet<string> DurableTriggers = new(StringComparer.OrdinalIgnoreCase)
+        {
+            "entityTrigger",
+            "activityTrigger",
+            "orchestrationTrigger"
+        };
+
+        /// <summary>
+        /// Checks if a <see cref="BindingMetadata"/> represents an HTTP trigger.
+        /// </summary>
+        /// <param name="binding">The binding metadata to check.</param>
+        /// <returns><c>true</c> if an HTTP trigger, <c>false</c> otherwise.</returns>
+        public static bool IsHttpTrigger(this BindingMetadata binding)
+        {
+            if (binding is null)
+            {
+                throw new ArgumentNullException(nameof(binding));
+            }
+
+            return string.Equals(HttpTriggerKey, binding.Type, StringComparison.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Checks if a <see cref="BindingMetadata"/> represents an webhook trigger.
+        /// </summary>
+        /// <param name="binding">The binding metadata to check.</param>
+        /// <returns><c>true</c> if a webhook trigger, <c>false</c> otherwise.</returns>
+        /// <remarks>
+        /// Known webhook triggers includes Event Grid and Event Grid sourced blob triggers.
+        /// </remarks>
+        public static bool IsWebHookTrigger(this BindingMetadata binding)
```

Comment:
Recommend adding tests for this method. In the tests, it looks like you have one negative test for blobTrigger, but not a positive one where source=EventGrid?